ls1intum/Artemis
# Quiz Reset button

### Checklist

### Motivation and Context


Fixed following issue: https://jirabruegge.in.tum.de/browse/ARTEMIS-361
You can now reset quiz exercises again.

### Description

SubmittedAnswers have a cascade remove relation with the QuizQuestions, meaning that when a submitted answer gets deleted the corresponding quiz questions get deleted as well. This is not the behaviour that we intend. Therefore we deleted this cascade remove relation.

### Steps for Testing


1. Log in to ArTEMiS
2. Navigate to Course Administration
3. Create a quiz exercise
4. Participate in that quiz exercise
5. Go back to Course administration
6. Go to the quiz exercise and click on the reset button
7. You should be able to have the same behaviour as after the creation of the quiz

### Screenshots



Reset quiz and check if question is still available.

![resetQuiz1](https://user-images.githubusercontent.com/41108487/61795469-c133da00-ae23-11e9-8f7f-0ad0e7e78a06.gif)

Start the quiz again, participate and check if participation is there

![resetQuiz12](https://user-images.githubusercontent.com/41108487/61795534-ede7f180-ae23-11e9-9d56-81d0b578a9f8.gif)

Participation is there and reset deletes again the participation (questions are still there -> first gif)

![resetQuiz2](https://user-images.githubusercontent.com/41108487/61820455-4c799380-ae55-11e9-8a60-345d17791d20.gif)
